syntax = "proto3";

option java_multiple_files = true;
option java_package = "pl.edu.agh.xdcs.api";

package xdcs.agent;

import "xdcs_api/common.proto";

enum ObjectType {
    BLOB = 0;
    TREE = 1;
}

message ObjectKey {
    string objectId = 1;
    ObjectType objectType = 2;
}

message ObjectIds {
    repeated string objectIds = 1;
}

message DependencyResolutionRequest {
    repeated ObjectKey objectKeys = 1;
    uint32 depth = 2;
}

service ObjectRepository {
    /**
     * Request objects, the requested objects will be sent as chunks,
     * compressed using the ZIP format.
     */
    rpc RequestObjects (stream ObjectIds) returns (stream Chunk) {

    }

    rpc ResolveDependencies (DependencyResolutionRequest) returns (ObjectIds) {

    }
}
